### ISSUE TITLE
Possible issue in client selection

### ISSUE THREAD

**Issue by zebengberg** [ecmwf/cdsapi#112]
The key for the now legacy CDS API [takes the form](https://cds.climate.copernicus.eu/api-how-to) `{uid}:{api-key}`. The new Beta CDS API [no longer contains](https://cds-beta.climate.copernicus.eu/how-to-api#install-the-cds-api-token) the `{uid}:` prefix. This new convention is [reaffirmed here](https://confluence.ecmwf.int/display/CKB/Please+read%3A+CDS+and+ADS+migrating+to+new+infrastructure%3A+Common+Data+Store+%28CDS%29+Engine).

Is the [client selection logic](https://github.com/ecmwf/cdsapi/blob/fe2ffc5f1afefbfe75584e5b36f4e832d1435270/cdsapi/api.py#L282-L286) in `Client.__new__` right? If I'm reading it correctly, the new client is chosen if `":"` appears in the token, otherwise the legacy client is chosen. Should this be reversed?

**Comment by lorenzo-solcast** [comment]
+1

**Comment by LeoBreebaart** [comment]
I arrived at the same conclusion after some inexplicable errors and subsequent debugging. 

Only the old CDS/ADS keys contain a `:`, the newer ones don't,  so that's when the new Client() should be created, but instead the logic is reversed and the legacy client is chosen:

```python
class Client(object):
    logger = logging.getLogger("cdsapi")

    def __new__(cls, url=None, key=None, *args, **kwargs):
        _, token, _ = get_url_key_verify(url, key, None)
        if ":" in token:
            return super().__new__(cls)
        import cads_api_client.legacy_api_client

        return super().__new__(cads_api_client.legacy_api_client.LegacyApiClient)
```

Since ECMWF is disabling the old API next week, I think it's really rather important that this gets fixed? Or am I missing something?

Basically, '+1' is what I'm saying here...

**Comment by gilcapote** [comment]
+1

**Comment by jreniel** [comment]
Possibly related to https://github.com/ecmwf/cdsapi/issues/131 ?

**Comment by joseph-sch** [comment]
The way I understand it, the `cdsapi.api.Client` delegates to `datapi` package's `LegacyApiClient` class when it detects the _new_ API credentials, as opposed to instantiating _itself_ (`cdsapi.api.Client`) when the _old_ API credentials are detected.

Within the `datapi` package, the `datapi.legacy_api_client.LegacyApiClient` uses instanciates a `datapi.api_client.ApiClient` object, so apparently it is just meant to expose a "legacy" Python interface to the current API client.

So, unless I'm mistaken, the code within `cdsapi` (namely the `cdsapi.api.Client` class) is not used anymore, except as a wrapper around `datapi.legacy_api_client.LegacyApiClient` which is itself a wrapper around `datapi.api_client.ApiClient`.

(My personal conclusion is to skip both `cdsapi` and the `LegacyApiClient`, and switch to using directly `datapi.ApiClient` for all my needs...)
